cmake/SetupSpheral.cmake

```diff
@@ -57,6 +57,12 @@ if(ENABLE_STATIC_CXXONLY)
   set(ENABLE_SHARED OFF)
 endif()
 
+if(ENABLE_CXXONLY)
+  set(ENABLE_INSTANTIATIONS OFF)
+  set(ENABLE_DOCS OFF)
+  set(ENABLE_TESTS OFF)
+endif()
+
```

Comment:
If we turn off ENABLE_INSTANTIATIONS, isn't this going to prevent building all the necessary code in the CXX library?  Implicit instantiation doesn't work if nothing is requesting that instantiation (like a main), so doesn't this cause a bunch of our code to be dropped from the final CXX library?

Reply:
I turned it off when building C++ because of the logic here: https://github.com/LLNL/spheral/blob/b8b8d41da4e46fcc6c0b8ea744e7d704756d706a/cmake/spheral/InstantiateCXX.cmake#L31

It looked to me that when ENABLE_INSTANTIATIONS was on that we needed python to generate the files, but in the else() the case without python in handled. Let me know if I'm misunderstanding something.

Reply:
Yep, but I think this will turn off most of our compiled code.  If you compare the size of libSpheral_CXX with and without this flag, is there a difference?

Reply:
Nope they are both 43M

Reply:
Both builds with without the ENABLE_INSTANTIATIONS and ENABLE_DOCS OFF, so if that's preferred I can remove those.